installing/installing_aws/installing-aws-customizations.adoc

```diff
@@ -44,6 +44,8 @@ include::modules/installation-configuration-parameters.adoc[leveloffset=+2]
 
 include::modules/installation-aws-config-yaml.adoc[leveloffset=+2]
 
+include::modules/installation-configure-proxy.adoc[leveloffset=+2]
```

Comment:
I have a doubt whether we should add it here. As I know, only UPI way support proxy setting until IPI supports using a preconfigured VPC. This section is for IPI-on-AWS.

Reply:
You _can_ use a proxy on AWS IPI, the proxy just has to live outside the VPC and still be reachable by the cluster.  This is how CI works now.  Not as nice as what's in flight with openshift/release#4719, but still something.

Reply:
@gpei Are you suggesting that this should not be included here for AWS IPI (and Azure and GCP)? I had confirmed with @katherinedube  the appropriate supported list here: https://jira.coreos.com/browse/OSDOCS-640?focusedCommentId=130922&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-130922

Reply:
As Trevor mentioned, until we support deploying to a pre-existing VPC (planned for the next release), the proxy must live outside the VPC and be reachable by the nodes. While I agree it's not ideal, I'm not sure we should be saying it doesn't work unless it *really* doesn't work.

Reply:
yeah, I understand that we _can_ use proxy for IPI install. But QE was told `UPI must be used for proxy testing` in 4.2, so all proxy related testing were scheduled with UPI installs in QE's 4.2 test matrix.

Reply:
All proxy CI to date has been IPI (with a pivot to UPI still in flight).

Reply:
I just checked through the OpenShift QE test matrix google doc and went through all the comments. I traced it back to a discussion Daneyon had with Xiaoli to remove the testing of ipi installs (at the time that it might have been the belief), but after ipi was tested, it wasn't the case. However, I doesn't look like anyone told QE of this. Everything including docs, CEE enablement, what we've told customers/field, etc. mentions that ipi will work. Can we get QE to do some testing on ipi installs?

Reply:
As @wking mentions, IPI can _technically_ be used to install a proxy-enabled cluster. However, do we believe this is how any customer would use the feature? If so, I'm interested in better understanding the use case. We took this approach for phase 1 of proxy CI out of necessity and with the understanding that phase 2 (UPI-based) is the real way of setting up a proxy-enabled cluster. If IPI proxy is included in the docs, we must make sure the reader understands that components are proxied only because they honor proxy env var settings and can simply bypass proxying by unsetting these env vars.

Reply:
> I traced it back to a discussion Daneyon had with Xiaoli to remove the testing of ipi installs (at the time that it might have been the belief), but after ipi was tested, it wasn't the case.

@katherinedube IPI was being used for testing as a stop-gap until UPI CI was complete. It was identified early on by @derekwaynecarr that UPI is required due to the lack of existing VPC support for IPI. As I mention in my above comment, IPI can technically work, but I don't see the requirements for IPI proxy meeting real world use cases.

Reply:
Just to be clear, even if we're going to say upi support is required for proxy this should only effect public cloud provider ipi deployments. Meaning I don't know this is really a generic ipi problem across all of our ipi providers like OSP. As it is today in the text matrix, it's all ipi providers - so nothing is being tested with OSP (not sure this should have been excluded.) Otherwise, why couldn't a public cloud provider use proxy with ipi? Could you provider some additional details as to why it wouldn't be practical? Thanks!

Reply:
I totally agree with @danehans 's statement.
IPI's prerequisite is having internet connectivity. That means, cluster itself already have internet connection, why still need proxy? Even proxy is enabled, how to validate all components and operators are really getting outside via proxy but not via its own internet connectivity? 
In QE's testing, we dropped internet connectivity, and enable proxy, we found the following bugs:
Bug 1753467 - [proxy] no proxy is set for kube-controller-manager

That indicates if we do not drop internet connectivity for proxy testing, our test result is NOT persuadable, or else, we would miss the above bugs. But in the whole IPI install process, we have no way to drop internet connectivity.

Reply:
> In QE's testing, we dropped internet connectivity, and enable proxy, we found...

We definitely want to support proxy on blackholed UPI; I don't think anyone opposes that.  Once we have bring-your-own-VPC, we all agree we want to support proxy there too.  And testing in *those* environments makes me reasonably confident that at least OpenShift components will respect a voluntary proxy.  The only contention seems to be whether we want to support proxy for installer-provided-everything.  And the pushback seems to be mostly around "who wants this?".  But I don't see anything about that case that would be a support burden, so I'd rather not add confusion by walling it off as an unsupported niche.  Maybe customers want a proxy for MitM snooping on our egress (and they are also convinced from our blackhole support that we are playing nice).  Or maybe they want a proxy for some other reason.  But I much prefer the optics of "you may not care about this on IPI" to "we don't support this (suggesting it does not work reliably) on IPI".  From a QE angle, we don't test all possible config permutations on all platforms.  I'm fine leaving IPI+proxy covered just in CI, and relying on QE in the blackholed cases to turn up bugs like they're already doing.

Reply:
>  From a QE angle, we don't test all possible config permutations on all platforms.
Yeah, that is my concern. QE did not run proxy testing in a blackholed IPI (due to in 4.2 IPI, we have no way to create a blackholed env), but only proxy testing in a blackholed UPI.

I like "we don't support this (suggesting it does not work reliably) on IPI".